File: src/main/java/jp/developer/bbee/pcassem/domain/auth/FirebaseIdTokenVerifier.java
package jp.developer.bbee.pcassem.domain.auth;

import com.google.firebase.FirebaseApp;
import com.google.firebase.auth.FirebaseAuth;
import org.springframework.stereotype.Component;

@Component
public class FirebaseIdTokenVerifier implements IdTokenVerifier {

    private final FirebaseApp firebaseApp;

    public FirebaseIdTokenVerifier(FirebaseApp firebaseApp) {
        this.firebaseApp = firebaseApp;
    }

    @Override
    public String verifyAndGetUid(String idToken) throws Exception {
        return FirebaseAuth.getInstance(firebaseApp).verifyIdToken(idToken).getUid();
    }
}

File: src/main/java/jp/developer/bbee/pcassem/domain/auth/FirebaseIdTokenVerifier.kt
package jp.developer.bbee.pcassem.domain.auth

import com.google.firebase.FirebaseApp
import com.google.firebase.auth.FirebaseAuth
import org.springframework.stereotype.Component

@Component
class FirebaseIdTokenVerifier(private val firebaseApp: FirebaseApp) : IdTokenVerifier {

    override fun verifyAndGetUid(idToken: String): String =
        FirebaseAuth.getInstance(firebaseApp).verifyIdToken(idToken).uid
}
